accesspoc/dips/tasks.py

```diff
@@ -0,0 +1,33 @@
+from celery import shared_task
+from .parsemets import METS
+
+import os
+import re
+import shutil
+import tempfile
+import zipfile
+
+
+@shared_task
+def extract_and_parse_mets(dip_id, zip_path):
+    """
+    Extracts a METS file from a given DIP zip file and uses the METS class
+    to parse its content, create the related DigitalFiles and update the DIP
+    DC metadata. Creates and deletes a temporary directory to hold the METS
+    file during its parsing. This function is meant to be called with
+    `.delay()` to be executed asynchronously by the Celery worker.
+    """
+    # Create temporary directory
+    tmpdir = tempfile.mkdtemp()
+    if not os.path.isdir(tmpdir):
+        os.mkdirs(tmpdir)
+    # Extract METS file
+    zip = zipfile.ZipFile(zip_path)
+    for info in zip.infolist():
+        if re.match(r'.*METS.[0-9a-f\-]{36}.*$', info.filename):
+            metsfile = zip.extract(info, tmpdir)
+    # Parse METS file
+    mets = METS(os.path.abspath(metsfile), dip_id)
+    mets.parse_mets()
```

Comment:
There's a bunch of database action going on here. I'm wondering if Celery wraps all that within a database transaction or you have to do it manually. Have you thought about that at all?

Also, introducing workers may be a reason to move to MySQL. Maybe. We should investigate further. See https://stackoverflow.com/a/10387821. Is WAL enabled for you?

Reply:
It looks like it has to be done manually, working on that ;)

This is interesting: https://www.sqlite.org/tempfiles.html

And Django only complains about the save points when "autocommit" is turned off: https://docs.djangoproject.com/en/1.11/topics/db/transactions/#savepoints-in-sqlite